ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/BlockSimulator.java

```diff
@@ -252,39 +314,40 @@ protected BlockCallSimulationResult processTransactions(
             "Transaction simulator result is invalid", transactionSimulationResult);
       }
       transactionUpdater.commit();
-      blockCallSimulationResult.add(transactionSimulationResult, ws);
+      blockStateCallSimulationResult.add(transactionSimulationResult, ws, operationTracer);
     }
-    return blockCallSimulationResult;
+    return blockStateCallSimulationResult;
   }
 
   private BlockSimulationResult createFinalBlock(
       final BlockHeader blockHeader,
-      final BlockCallSimulationResult blockCallSimulationResult,
+      final BlockStateCallSimulationResult blockStateCallSimulationResult,
       final BlockOverrides blockOverrides,
-      final MutableWorldState ws) {
+      final MutableWorldState ws,
+      final Optional<List<Request>> maybeRequests) {
 
-    List<Transaction> transactions = blockCallSimulationResult.getTransactions();
-    List<TransactionReceipt> receipts = blockCallSimulationResult.getReceipts();
+    List<Transaction> transactions = blockStateCallSimulationResult.getTransactions();
+    List<TransactionReceipt> receipts = blockStateCallSimulationResult.getReceipts();
 
     BlockHeader finalBlockHeader =
         BlockHeaderBuilder.createDefault()
             .populateFrom(blockHeader)
             .ommersHash(BodyValidation.ommersHash(List.of()))
-            .stateRoot(blockOverrides.getStateRoot().orElseGet(ws::rootHash))
+            .stateRoot(blockOverrides.getStateRoot().orElseGet(ws::frontierRootHash))
```

Comment:
why `frontierRootHash`?

Reply:
@fab-10 Since `eth_simulationV1` can simulate up to 256 blocks in the future, I need to retain the changes from each block in the accumulator to build the next block, while also needing the state root to construct the block header. The Frontier Root Hash does not clear the accumulator as the rootHash method does. One option could be to rename it to "intermediateRootHash." Perhaps @matkt has a better suggestion on how I could achieve this?

Reply:
why you need to keep what you have in the accumulator ? you just have to persist in your worldstate for each block and start with empty accumulator for each block . no ?